pymeasure/instruments/instrument.py

```diff
@@ -528,3 +528,13 @@ def check_errors(self):
             return errors
         else:
             raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
+
+    def next_error(self):
+        """ Read the next error from the instrument.
+
+        :return: error entry
+        """
```

Comment:
I suggest making this a private method, as users should always use  `check_errors()` for consistency.

```suggestion
    def _next_error(self):
        """ Read the next error from the instrument.
        
        Internal method -- use :meth:`check_errors` to query for error status.
        Reimplement this method in instruments using a different command.

        :return: error entry
        """
```

Reply:
In my opinion, `next_error` is consistent with all the methods already present for SCPI commands. On the other hand `check_errors` is a utility function that extract all the errors and log them.
There could be several use cases:
- SCPI instruments: retrieve all errors using check_errors.
- SCPI instruments: you just need next error, use next_error
- Non SCPI instruments: customize next_error or check_errors or both as needed.

Reply:
To provide context: I want to make sure we don't add public API to _all_ our instruments without a clear use case for it. I have seen in other code bases how messy an unchecked proliferation of public API in an inheritance tree can become, and I _very much_ want to avoid that here.